### PR TITLE
[Rename] Update README.markdown with links to OpenSearch documentation

### PR DESCRIPTION
**Pull request by harold-wang**
### Description
[Describe what this change achieves]
Update elastic.co related link docs
 
### Issues Resolved
[List any issues this PR will resolve]
#70 , #360
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by harold-wang** [comment]
This is replacement of #517 (#517 was not reachable now)

**Comment by adnapibar** [comment]
I don't think the docs link are not ready yet - please do not merge until they do.

**Comment by harold-wang** [comment]
start gradle check

**Comment by harold-wang** [comment]
> I don't think the docs link are ready yet - please do not merge until they are.

This PR is replacement of #517, #517 was not accessible in April 12, 2021, will not be merged until document is ready as discussed in #517.

File: rest-api-spec/README.markdown [hunk 1]
```diff
@@ -4,13 +4,13 @@ This repository contains a collection of JSON files which describe the [OpenSear
 
 Their purpose is to formalize and standardize the API, to facilitate development of libraries and integrations.
 
-Example for the ["Create Index"](https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-create-index.html) API:
+Example for the ["Create Index"](https://www.opensearch.org/guide/en/openearch/reference/master/indices-create-index.html) API:
```
Comment (carlspring) [inline rest-api-spec/README.markdown:7]: There is a typo in the URL.

Comment (harold-wang) [reply rest-api-spec/README.markdown]: > There is a typo in the URL.

Fixed, thanks

**Comment by peterzhuamazon** [comment]
start dco check
start wrapper validation
start gradle precommit

**Comment by peterzhuamazon** [comment]
start gradle precommit

**Pull request closed without merging**